src/App.tsx

```diff
@@ -1,204 +1,97 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useRef,
+  useMemo,
+  useCallback,
+} from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState<string | null>(null);
+  const [filter, setFilter] = useState<Filter>('all');
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setIsError('Unable to load todos');
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (isError) {
+      timerRef.current = setTimeout(() => {
+        setIsError(null);
+      }, 3000);
+
+      return () => {
+        if (timerRef.current) {
+          clearTimeout(timerRef.current);
+        }
+      };
+    }
+
+    return () => {};
+  }, [isError]);
```

Comment:
clearTimeout should always run when the effect is cleaned up and second return is redundant 
```suggestion
  useEffect(() => {
    if (isError) {
      timerRef.current = setTimeout(() => {
        setIsError(null);
      }, 3000);
    }
    
    return () => {
      if (timerRef.current) {
        clearTimeout(timerRef.current);
      }
    };
  }, [isError]);
```